### PR TITLE
FAI-12542: Pin Dockerfile.jdbc to eclipse-temurin:21-jdk-jammy

### PR DESCRIPTION
**Pull request by eskrm**

### CODE REVIEW

File: Dockerfile.jdbc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:11 AS builder
+FROM eclipse-temurin:21-jdk-jammy AS builder
```
Comment (eskrm) [inline Dockerfile.jdbc:1]: 1. Pin to Ubuntu Jammy so gcc matches the version in the Metabase image.
2. Likewise, use JDK 21 since the Metabase image uses JRE 21. This is unrelated to the glibc errors.

**Review by tovbinm — APPROVED** [review]

File: Dockerfile.jdbc (hunk 1 above)
Comment (tovbinm) [reply Dockerfile.jdbc:1]: 2. why eclipse-temurin and not zulu-openjdk-alpine ?

Comment (eskrm) [reply Dockerfile.jdbc:1]: https://whichjdk.com/

Also, we can't build on alpine. DuckDB needs gcc/glibc.
